test/new-e2e/tests/containers/k8s_test.go

```diff
@@ -247,6 +248,27 @@ func (suite *k8sSuite) TestVersion() {
 	}
 }
 
+func (suite *k8sSuite) TestClusterAgentConfigCheck() {
+	ctx := context.Background()
+	suite.Run("cluster agent pods return the right information for the config check command", func() {
+		appSelector := suite.AgentLinuxHelmInstallName + "-datadog-cluster-agent"
+
+		linuxPods, err := suite.K8sClient.CoreV1().Pods("datadog").List(ctx, metav1.ListOptions{
+			LabelSelector: fields.OneTermEqualSelector("app", appSelector).String(),
+			Limit:         1,
+		})
+		if suite.NoError(err) && len(linuxPods.Items) >= 1 {
+			stdout, stderr, err := suite.podExec("datadog", linuxPods.Items[0].Name, "cluster-agent", []string{"agent", "configcheck"})
+			if suite.NoError(err) {
+				suite.Empty(stderr, "Standard error of `agent configcheck` should be empty")
+				suite.Contains(stdout, "=== kubernetes_apiserver check ===")
+			} else {
+				suite.T().Errorf("No cluster agent pod found. %v\n", err)
+			}
+		}
```

Comment:
We can leverage the fact that the `require` package makes the test fail early and return as soon as an error is encountered to get rid of the nested `if`.

```suggestion
		suite.Require().NoError(err)
		suite.Require().Len(linuxPods.Items, 1)
		stdout, stderr, err := suite.podExec("datadog", linuxPods.Items[0].Name, "cluster-agent", []string{"agent", "configcheck"})
		suite.Require().NoError(err)
		suite.Empty(stderr, "Standard error of `agent configcheck` should be empty")
		suite.Contains(stdout, "=== kubernetes_apiserver check ===")
```